apps/web/vibes/soul/types/cart.ts

```diff
@@ -0,0 +1,47 @@
+import { Image } from './image'
+
+export interface CartSummary {
+  title?: string
+  caption?: string
+  subtotalLabel?: string
+  subtotal: string | Promise<string>
+  shippingLabel?: string
+  shipping?: string
+  taxLabel?: string
+  tax?: string | Promise<string>
+  grandTotalLabel?: string
+  grandTotal?: string | Promise<string>
+  ctaLabel?: string
+}
+
+export interface CartProps {
+  title?: string
+  lineItems: CartLineItem[] | Promise<CartLineItem[]>
+  summary: CartSummary
+  emptyState: CartEmptyState
+  removeItemAriaLabel?: string
+  loadingAriaLabel?: string
+  removeLineItemAction(id: string): Promise<void>
```

Comment:
FYI, we're going to need to update these actions to that they follow the interface expected by `useActionState`:

```ts
type Action<State, Payload> = (state: State, payload: Payload) => State | Promise<State>
```

A lot of this will have a state of `{ error: string | null }` so that we can handle error alerts in VIBES

Reply:
Gotcha. feels like I should wait until that code is pushed though right? Then come back and update?